modules/account-lib/src/coin/dot/singletonRegistry.ts

```diff
@@ -0,0 +1,24 @@
+import { TypeRegistry } from '@substrate/txwrapper-core/lib/types';
+import { getRegistry } from '@substrate/txwrapper-polkadot';
+import { PolkadotSpecNameType } from '@bitgo/statics';
+import { Metadata } from '@polkadot/types/metadata/Metadata';
+import { Material } from './iface';
+
+export class SingletonRegistry {
+  private static instance: TypeRegistry;
+  private static material: Material;
+  private static metadata: Metadata;
+
+  static getInstance(material): TypeRegistry {
+    if (material !== SingletonRegistry.material) {
+      SingletonRegistry.material = material;
```

Comment:
ditto on the reducing memory here: Since we are storing the material and the instance at the same time, it seems like we are duplicating and persisting the 2 big strings in memory again. This seems like a caching effort to reduce computation?